### PR TITLE
Don't generate minification variables if not needed

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Follow up of #59158 and #59157.

r? @QuietMisdreavus 

cc @jdm

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
I wonder if we can make a test for this. Can we use `rustdoc` tests to look at the search index javascript? Since the initial minification variables are always going to have the same line to initialize them, we could look for something like that. I think we can also use an `auxiliary` crate to test what happens when a dependency has minified the search index but a downstream crate has disabled it - [from what i can tell](https://github.com/rust-lang/rust/blob/54479c624cc24bb852291d83e11cc19389dcb21b/src/tools/compiletest/src/runtest.rs#L1432-L1434), auxiliary doc builds use the properties from the aux file itself instead of the main test.

**Comment by GuillaumeGomez** [comment]
I could write a test which would simply load the searchIndex file and if an error occurs, it means the generation failed. However, this would require a new test-suite since we need to build using `--disable-minification`.

**Comment by QuietMisdreavus** [comment]
I was thinking something even simpler, doing a substring search using a `@has` rule in a regular rustdoc test. We can pass the `--disable-minification` flag in those just fine - a `compile-flags:--disable-minification -Z unstable-options` rule in the test should suffice.

**Comment by GuillaumeGomez** [comment]
Such a test would completely fail to trigger when a minification would appear. :-/

**Comment by ollie27** [comment]
Is this added complexity worth it to save a few bytes in the none minified `search-index.js`?

**Comment by GuillaumeGomez** [comment]
This is a simple change which generates less things. For me, it's totally worth it, although we can't really test it (or we'll need to create a whole new test suite, which I'd prefer to avoid).

**Comment by Mark-Simulacrum** [comment]
@QuietMisdreavus Could we get a review/decision here either way?

I'm personally also somewhat inclined to say that we shouldn't do this as the 10-20 bytes we're saving aren't really worth the complexity introduced here in my opinion.

**Comment by GuillaumeGomez** [comment]
Easy to say when you're not the one debugging the minified JS. :p

**Comment by GuillaumeGomez** [comment]
Oh also, it fixes a bug currently present in the code, it doesn't add a new feature. Therefore it seems pretty worth it (but my opinion is completely biased in here since I use this feature quite heavily).

**Comment by Mark-Simulacrum** [comment]
Ah, I was not aware that this fixes a bug -- I presumed that the variables don't hurt if they're unused, but I guess that's not true.

**Comment by GuillaumeGomez** [comment]
By default, people don't have this bug since it needs a nightly-only option to be enabled. So indeed, it doesn't hurt to have unused variables if they're really not used. However, in here in some cases, the code references to them.

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -964,10 +967,11 @@ themePicker.onblur = handleThemeButtonsBlur;
                     variables.push(line.clone());
                     continue;
                 }
-                if !line.starts_with(key) {
+                if !line.starts_with(key) || line.starts_with(&format!(r#"{}["{}"]"#, key, krate)) {
                     continue;
                 }
-                if line.starts_with(&format!(r#"{}["{}"]"#, key, krate)) {
+                if line.starts_with(MINIFIED_VARIABLES) {
+                    is_minified = true;
```
Comment (ollie27) [inline src/librustdoc/html/render.rs]: This is dead code. The above:

```rust
if !line.starts_with(key) || line.starts_with(&format!(r#"{}["{}"]"#, key, krate)) {
    continue;
}
```

means that `line.starts_with(MINIFIED_VARIABLES)` can never be true. This means that the minification variables are missing if the last call to rustdoc used `--disable-minification` but previous ones didn't.

**Review by ollie27 — CHANGES_REQUESTED** [review]

**Comment by ollie27** [comment]
@bors r-

Again I don't think this added complexity is worth it. Especially when it doesn't even work and has no tests.

**Comment by joelpalmer** [comment]
Hi @GuillaumeGomez this is a ping from Triage. Please post an update on this PR at your earliest convenience.

**Comment by GuillaumeGomez** [comment]
@ollie27 is against it so I guess I'll just close it...

**Pull request closed without merging**